tinygrad/tensor.py

```diff
@@ -2700,8 +2700,8 @@ def scaled_dot_product_attention(self, key:Tensor, value:Tensor, attn_mask:Optio
     assert all_int(self.shape), f"does not support symbolic shape {self.shape}"
     if is_causal: attn_mask = Tensor.ones(self.shape[-2], key.shape[-2], requires_grad=False, device=self.device).tril(0).cast(dtypes.bool)
     if attn_mask is not None and attn_mask.dtype == dtypes.bool: attn_mask = (attn_mask == 0).where(-float("inf"), 0)
-    qk = self @ key.transpose(-2,-1) / math.sqrt(self.shape[-1])
-    return ((qk+attn_mask) if attn_mask is not None else qk).softmax(-1).dropout(dropout_p) @ value
+    qk = self.matmul(key.transpose(-2,-1), acc_dtype=dtypes.float32) / math.sqrt(self.shape[-1])
```

Comment:
`acc_dtype=least_upper_dtype(self.dtype, key.dtype, dtypes.float32)` so that float64 is not downcasted.